### PR TITLE
Improve logging of test for panel destruction

### PR DESCRIPTION
**Pull request by feerrenrut**
### Link to issue number:
None

### Summary of the issue:
While looking into #12220, I noticed a lot of logging noise for the tests for destruction of settings panels.

A little investigation showed that the same settings dialogs was being marked as destroyed previously because the `evt.skip()`
was not called.
Calling `skip` is not appropriate, but it is also not necessary to log that the dialog which is already marked as destoyed, will be again.

### Description of how this pull request fixes the issue:
Use an enum for destroyed state, this logs better.
Only log if the dialog is not already marked as destoyed.
Improve what is logged so that the dialog is actually identifiable.

Example log message after PR:
```
Input: kb(desktop):escape
DEBUG - gui.settingsDialogs.SettingsDialog._setInstanceDestroyedState (18:46:00.444) - MainThread (15748):
Setting state to destroyed for instance: Input Gestures - InputGesturesDialog - <gui.inputGestures.InputGesturesDialog object at 0x1AB17DA8>
Current _instances ['NVDA Settings - CREATED', 'Input Gestures - CREATED']
IO - speech.speak (18:46:00.517) - MainThread (15748):
Speaking [LangChangeCommand ('en'), 'NVDA Settings: General (normal configuration)', 'dialog', CancellableSpeech (still valid)]
IO - speech.speak (18:46:00.519) - MainThread (15748):
Speaking [LangChangeCommand ('en'), 'Categories:', 'list', CancellableSpeech (still valid)]
IO - speech.speak (18:46:00.521) - MainThread (15748):
Speaking [LangChangeCommand ('en'), 'General', '1 of 13', CancellableSpeech (still valid)]
IO - inputCore.InputManager.executeGesture (18:46:00.802) - winInputHook (12364):
Input: kb(desktop):escape
DEBUG - gui.settingsDialogs.SettingsDialog._setInstanceDestroyedState (18:46:00.804) - MainThread (15748):
Setting state to destroyed for instance: NVDA Settings - NVDASettingsDialog - <gui.settingsDialogs.NVDASettingsDialog object at 0x1A8F8100>
Current _instances ['NVDA Settings - CREATED']
```

### Testing strategy:
Ran locally with the NVDA settings dialog, checked log.

### Known issues with pull request:
None

### Change log entry:
None, these were internal constants, meant for debugging purposes only.

### Code Review Checklist:

This checklist is a reminder of things commonly forgotten in a new PR.
Authors, please do a self-review and confirm you have considered the following items.
Mark items you have considered by checking them.
You can do this when editing the Pull request description with an x: `[ ]` becomes `[x]`.
You can also check the checkboxes after the PR is created.

- [x] Pull Request description is up to date.
- [x] Unit tests.
- [x] System (end to end) tests.
- [x] Manual tests.
- [x] User Documentation.
- [x] Change log entry.
- [x] Context sensitive help for GUI changes.

### CODE REVIEW

File: source/_UIAHandler.py [hunk 1]
```diff
@@ -37,7 +37,7 @@
 import UIAUtils
 from comInterfaces import UIAutomationClient as UIA
 # F403: unable to detect undefined names
-from comInterfaces .UIAutomationClient import *  # noqa:  F403
+from comInterfaces.UIAutomationClient import *  # noqa:  F403
```
Comment (lukaszgo1) [inline source/_UIAHandler.py:40]: Why this is in the diff?

**Review by michaelDCurran — APPROVED** [review]

File: source/gui/__init__.py [hunk 2]
```diff
@@ -585,10 +587,16 @@ def terminate():
...
+	# prevent deletion of the object while we work on it.
+	_SettingsDialog = settingsDialogs.SettingsDialog
+	nonWeak: typing.Dict[_SettingsDialog, _SettingsDialog] = dict(_SettingsDialog._instances)
+
+	for instance, state in nonWeak.items():
+		if state is _SettingsDialog.DialogState.DESTROYED:
 			log.error(
-				"Destroyed but not deleted instance of settings dialog exists: {!r}".format(instance)
+				"Destroyed but not deleted instance of gui.SettingsDialog exists"
+				f": {instance.title} - {instance.__class__.__qualname__} - {instance}"
```
Comment (seanbudd) [inline source/gui/__init__.py:599]: Just a heads up that I've removed this code entirely in https://github.com/nvaccess/nvda/pull/12286

Comment (feerrenrut) [reply source/gui/__init__.py:599]: I think that code is still useful, at least in theory. It warns us that we are holding onto a python object even though the corresponding wx.Window has already been destroyed. For some derivatives of SettingsDialog, there should only be one instance, so we keep track of the objects. In the past we have occasionally run into issues where objects are kept alive. This helps us to get an early warning during development. I'd be happy for it to be moved into a helper function with a comment that explains this though.

File: source/_UIAHandler.py (hunk 1 above)
Comment (feerrenrut) [reply source/_UIAHandler.py:40]: There was a space after comInterfaces which looks strange. Apparently python doesn't care, but I would like to fix it anyway.

**Pull request merged**